pkg/api/common/versions.go

```diff
@@ -108,19 +108,13 @@ func GetAllSupportedKubernetesVersions() []string {
 func GetVersionsGt(versions []string, version string, inclusive, preReleases bool) []string {
 	// Try to get latest version matching the release
 	var ret []string
-	var cons *semver.Constraints
-	minVersion, _ := semver.NewVersion(version)
+	minVersion, _ := semver.Make(version)
 	for _, v := range versions {
-		sv, _ := semver.NewVersion(v)
-		if preReleases && minVersion.Prerelease() == "" {
-			sv, _ = semver.NewVersion(fmt.Sprintf("%d.%d.%d", sv.Major(), sv.Minor(), sv.Patch()))
+		sv, _ := semver.Make(v)
+		if !preReleases && len(sv.Pre) != 0 {
```

Comment:
if `preReleases` is false we don't want to include pre-releases so we skip any version with a pre-release part